homeassistant/components/geo_json_events/geo_location.py

```diff
@@ -105,7 +105,12 @@ async def async_update(self) -> None:
     def _update_from_feed(self, feed_entry: GenericFeedEntry) -> None:
         """Update the internal state from the provided feed entry."""
         if feed_entry.properties and "name" in feed_entry.properties:
-            self._attr_name = feed_entry.properties.get("name")
+            name_property = feed_entry.properties["name"]
+            if isinstance(name_property, int):
+                # Convert integer to string to avoid slugify issues with the entity name
+                self._attr_name = str(name_property)
+            else:
+                self._attr_name = name_property
```

Comment:
```suggestion
            # The entry name's type can vary, but our own name must be a string
            self._attr_name = str(feed_entry.properties["name"])
```
This seems more defensive to me and less likely to break in the future.

Reply:
Thank you - this is much cleaner and covers other scenarios too.